### PR TITLE
Fix ReactEditText so it works with Android Emoji2 automatic support

### PR DESCRIPTION
**Pull request by designedbyz**
## Summary
tldr; `ReactEditText` and Android's emoji support in Android's AppCompat 1.4.0 / 1.4.x conflict in an odd way, causing NPEs.

`ReactEditText` defines an `InternalKeyListener`, `mKeyListener`, that it uses to make sure input from all keyboards works correctly. This listener is normally initialized at the end of the constructor. 

Unfortunately, some versions of `AppCompatEditText`, most notably the version in the App Compat `1.4.0-alpha0x`, the [minimum version required for the Play Store's emoji compliance](https://developer.android.com/guide/topics/ui/look-and-feel/emoji2#appcompat) call  `setInputType` from `AppCompatEditText`'s constructor. `ReactEditText` operates on the key listener inside of `setInputType` and since the `AppCompatEditText` constructor is called via call to `super` before the key listener is initialized, these versions of app compat can cause NPEs when used with React Native. 

The fix is simple; check to see if `mKeyListener` is null, and initialize it if it is. This is necessary in both the constructor and inside of `setInputType`.

## Changelog

https://github.com/facebook/react-native/wiki/Changelog

[Android] [Fixed] - NPE in `ReactEditText.setInputType` when React Native is used with some versions of a AppCompat 1.4.x. (and possibly others)

## Test Plan

1. Build an app with both React Native and load it inside an app that is using AppCompat 1.4.x
2. Add a text field using React Native. 
3. Open the screen of the app that contains the text field. 

If you're working from this branch, you'll be fine. If you're working from main you'll get an NPE.

I can put together a test repo if needed.

### CODE REVIEW

**Comment by Dstiers11091** [comment]
jump

**Comment by designedbyz** [comment]
No clue why that many tests all failed to run for that change, but happy to rebase on latest or whatever is needed.

**Comment by cortinico** [comment]
Code looks good to me. 
Could you rebase as you're far behind main.

**Comment by designedbyz** [comment]
> Code looks good to me.

Awesome, thanks!

> Could you rebase as you're far behind main.

Haha, the distance is about the length of time I was waiting for the CLA to be signed, lol. Done!
